apis/refs/v1beta1/computerefs.go

```diff
@@ -27,6 +27,15 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// fixStaleExternalFormat converts the "External" reference field to the right format if a SelfLink value is used.
+// This guarantees the backward compatibility for Compute Beta resources.
+func fixStaleExternalFormat(external string) string {
+	external = strings.TrimPrefix(external, "https://www.googleapis.com/compute/v1/")
+	external = strings.TrimPrefix(external, "https://www.googleapis.com/compute/v1beta1/")
+	external = strings.TrimPrefix(external, "/")
+	return external
+}
+
 type ComputeNetworkRef struct {
 	/* The compute network selflink of form "projects/<project>/global/networks/<network>", when not managed by Config Connector. */
```

Comment:
Shall we update the comment? 

"projects/<project>/global/networks/<network>" is not a selfLink, it's a preferred value of KCC. We can simply say "The external compute network of form "projects/<project>/global/networks/<network>"" or similar.

Reply:
Good catch.